packages/eslint-plugin/src/util/misc.ts

```diff
@@ -4,12 +4,23 @@
 
 import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/utils';
 import { requiresQuoting } from '@typescript-eslint/type-utils';
+import * as ts from 'typescript';
 
+const DEFINITION_EXTENSIONS = [
+  ts.Extension.Dts,
+  ts.Extension.Dcts,
+  ts.Extension.Dmts,
+] as const;
 /**
  * Check if the context file name is *.d.ts or *.d.tsx
  */
 function isDefinitionFile(fileName: string): boolean {
-  return /\.d\.tsx?$/i.test(fileName || '');
+  for (const definitionExt of DEFINITION_EXTENSIONS) {
+    if (fileName.endsWith(definitionExt)) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
Nit: you could `return DEFINITION_EXTENSIONS.some(extension => fileName.endsWith(extension))`. If you wanted.